src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -81,73 +81,24 @@ public int showAllTaskEvents(){
 					deadlineTaskCounter++;
 				}
 			}
-
-			LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
-
-			if(!currentDate.equals(tempDate)){
-				currentDate = tempDate;
-				addLabel(currentDate);
+			
+			LocalDate itemToAddDate = LocalDate.MIN;
+			if(itemToAdd instanceof Event){
+				itemToAddDate = ((Event) itemToAdd).getStartDateTime().toLocalDate();
 			}
-
-			addedCounter++;
-			addTaskEvent(itemToAdd);
-		}
-
-		for(FloatingTask task: dataList.getFloatingTasksList()){
-			addedCounter++;
-			addTaskEvent(task);
-		}
-
-		return addedCounter;
-	}
-
-	public int showTaskEventsFromDate(LocalDate selectedDate){
-		LocalDate currentDate = null;
-		int eventCounter = 0;
-		int deadlineTaskCounter = 0;
-		int addedCounter = 0;
-
-		List<Event> eventsList = dataList.getEventsList();
-		List<DeadlineTask> deadlineTasksList = dataList.getDeadlineTasksList();
-
-		while ((eventCounter < eventsList.size()) || (deadlineTaskCounter < deadlineTasksList.size())) {
-			TaskEvent itemToAdd;
-
-			if (eventCounter >= eventsList.size()) {
-				itemToAdd = deadlineTasksList.get(deadlineTaskCounter);
-				deadlineTaskCounter++;
-			} else if (deadlineTaskCounter >= deadlineTasksList.size()) {
-				itemToAdd = eventsList.get(eventCounter);
-				eventCounter++;
-			} else {
-				Event nextEvent = eventsList.get(eventCounter);
-				LocalDate nextEventDate = nextEvent.getStartDateTime().toLocalDate();
-				DeadlineTask nextDeadlineTask = deadlineTasksList.get(deadlineTaskCounter);
-				LocalDate nextDeadlineTaskDate = nextDeadlineTask.getDateTime().toLocalDate();
-
-				if(!nextEventDate.isAfter(nextDeadlineTaskDate)){
-					itemToAdd = nextEvent;
-					eventCounter++;
-				} else {
-					itemToAdd = nextDeadlineTask;
-					deadlineTaskCounter++;
-				}
+			else if(itemToAdd instanceof DeadlineTask){
+				itemToAddDate = ((DeadlineTask) itemToAdd).getDateTime().toLocalDate();
 			}
 
-			if(checkNotBefore(selectedDate, itemToAdd)){
-				LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
-
-				if(currentDate == null){
-					currentDate = tempDate;
-
-					if(!selectedDate.equals(currentDate)){
+			if(selectedDate.isBefore(itemToAddDate)){
+				if(!currentDate.equals(itemToAddDate)){
+					if(currentDate.equals(LocalDate.MIN) && !selectedDate.equals(LocalDate.MIN) 
```

Comment:
checking equals to LocalDate.MIN is a bit hacky-ish as it does not tell the meaning. Basically you want to check whether you want to force display the selected date even if there are no items there. Just use a boolean as a function parameters. Also maintain a boolean to keep track of whether you have taken care of selectedDate